### PR TITLE
M3 io fixes

### PR DESCRIPTION
**Pull request by eatkins**
- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

### CODE REVIEW

**Comment by eatkins** [comment]
This resolves a number of issues:
#4651 
#4650 
#4649 
#4648 
#241

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1609/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/4664/head-develop-pull/238/head-develop-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1609/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/4664">pull/4664/head      </A> | sbt/sbt@5b49eabe4688df9c497f7121e9d6603f724e28f7
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@b6fc35f27615f129f91de9ac14400101fb8b9b7e
  io                | <A HREF="https://github.com/sbt/io/pull/238">pull/238/head       </A> | sbt/io@05ff469a5db5019e6d02b5a3a8798d78e2909cc8
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/develop">develop             </A> | sbt/librarymanagement@faabc7712317c0693e93173edd49e5b4ccfbb671
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@6e9a8241d43fb1f2502def35ee884526b3a3b691

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=pull/4664/head&zinc=develop&io=pull/238/head&librarymanagement=develop&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1610/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/4664/head-develop-pull/238/head-develop-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1610/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/4664">pull/4664/head      </A> | sbt/sbt@182a7b0e5f0bddb3b51409a5fc042700d768dacf
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@bbbb58e79565f8f23ab9474d061909292f1575d9
  io                | <A HREF="https://github.com/sbt/io/pull/238">pull/238/head       </A> | sbt/io@05ff469a5db5019e6d02b5a3a8798d78e2909cc8
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/develop">develop             </A> | sbt/librarymanagement@faabc7712317c0693e93173edd49e5b4ccfbb671
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@6e9a8241d43fb1f2502def35ee884526b3a3b691

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=pull/4664/head&zinc=develop&io=pull/238/head&librarymanagement=develop&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1611/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/4664/head-develop-pull/238/head-develop-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1611/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/4664">pull/4664/head      </A> | sbt/sbt@182a7b0e5f0bddb3b51409a5fc042700d768dacf
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@bbbb58e79565f8f23ab9474d061909292f1575d9
  io                | <A HREF="https://github.com/sbt/io/pull/238">pull/238/head       </A> | sbt/io@05ff469a5db5019e6d02b5a3a8798d78e2909cc8
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/develop">develop             </A> | sbt/librarymanagement@faabc7712317c0693e93173edd49e5b4ccfbb671
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@6e9a8241d43fb1f2502def35ee884526b3a3b691

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=pull/4664/head&zinc=develop&io=pull/238/head&librarymanagement=develop&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1612/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/4664/head-develop-pull/238/head-develop-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1612/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/4664">pull/4664/head      </A> | sbt/sbt@182a7b0e5f0bddb3b51409a5fc042700d768dacf
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@bbbb58e79565f8f23ab9474d061909292f1575d9
  io                | <A HREF="https://github.com/sbt/io/pull/238">pull/238/head       </A> | sbt/io@05ff469a5db5019e6d02b5a3a8798d78e2909cc8
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/develop">develop             </A> | sbt/librarymanagement@faabc7712317c0693e93173edd49e5b4ccfbb671
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@6e9a8241d43fb1f2502def35ee884526b3a3b691

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=pull/4664/head&zinc=develop&io=pull/238/head&librarymanagement=develop&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1613/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: pull/4664/head-develop-pull/238/head-develop-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1613/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/4664">pull/4664/head      </A> | sbt/sbt@182a7b0e5f0bddb3b51409a5fc042700d768dacf
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@bbbb58e79565f8f23ab9474d061909292f1575d9
  io                | <A HREF="https://github.com/sbt/io/pull/238">pull/238/head       </A> | sbt/io@05ff469a5db5019e6d02b5a3a8798d78e2909cc8
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/develop">develop             </A> | sbt/librarymanagement@faabc7712317c0693e93173edd49e5b4ccfbb671
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@6e9a8241d43fb1f2502def35ee884526b3a3b691

:white_check_mark: The result is: ***SUCCESS***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=pull/4664/head&zinc=develop&io=pull/238/head&librarymanagement=develop&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

File: main/src/main/scala/sbt/internal/nio/CheckMetaBuild.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package sbt
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/nio/CheckMetaBuild.scala]: Missing file header.

**Review by eed3si9n — APPROVED** [review]
Thanks for all the fixes!

**Pull request merged**